django/cantusdb_project/next_chants.py

```diff
@@ -14,11 +14,25 @@ def next_chants(cantus_id):
     concordances = Chant.objects.filter(cantus_id=cantus_id).only(
         "source", "folio", "sequence_number"
     )
-    next_chants = [chant.next_chant for chant in concordances]
+    # if not display_unpublished:
+    #     concordances = concordances.filter(source__published=True)
+    
+    next_chants = [chant.next_chant
+            for chant
+            in concordances
+            if chant.next_chant is not None
+        ]
+    if not display_unpublished:
```

Comment:
Just curious, but is there a reason you did it this way rather than with the commented out lines (17, 18) ? 

Not sure those commented lines are useful now

Reply:
thanks for catching this! I've changed it to use the version that was commented out. Either approach works (since a chant's next_chant will necessarily have the same source), but this one is simpler.